pyppeteer/connection.py

```diff
@@ -243,6 +243,15 @@ def send(self, method: str, params: dict = None) -> Awaitable:
                 ))
         return callback
 
+    async def send(self, method: str, params: dict = None) -> Awaitable:
+        future = self.send_unshield(method, params)
+        try:
+            return await asyncio.shield(future)
+        except asyncio.CancelledError:
+            if not future.done():
+                await future
```

Comment:
I guess this could be simplified to:
```suggestion
            await future
```
But adding the `if` helps for understanding :+1:

Reply:
You are right it's probably not needed.
But it fell natural when writing it, and it probably helps a bit to understand what is going on indeed.